### PR TITLE
CLI should maintain connection in case of agent restart

### PR DESCRIPTION
**Pull request by khagesh**
Agent will maintain connections if connection breaks due to some issue. Replaced `SimpleStack` to `KITStack`.

Tests worked fine in `bulldog` repo as well.

### CODE REVIEW

File: sovrin_client/agent/agent.py [hunk 1]
```diff
@@ -264,4 +264,4 @@ def isSchemaFound(schema, ranViaScript):
         if ranViaScript:
             exit(1)
         else:
-            raise Exception
\ No newline at end of file
+            raise Exception
```
Comment (jasonalaw) [inline sovrin_client/agent/agent.py]: It's good that we added the hard return, but the "raise Exception" is eating at me. We should at least raise a RuntimeError with a string.

File: sovrin_client/agent/endpoint.py [hunk 2]
```diff
@@ -29,7 +29,8 @@ def __init__(self, port: int, msgHandler: Callable,
         if basedirpath:
             stackParams["basedirpath"] = basedirpath
 
-        super().__init__(stackParams, self.baseMsgHandler)
+        self.registry = {}
+        super().__init__(stackParams, self.baseMsgHandler, self.registry)
```
Comment (jasonalaw) [inline sovrin_client/agent/endpoint.py]: The two lines above can be done with:
+        super().__init__(stackParams, self.baseMsgHandler, {})

The super init assigns self.registry.

File: sovrin_client/agent/endpoint.py [hunk 3]
```diff
@@ -62,8 +63,10 @@ def baseMsgHandler(self, msg):
...
-            # updates the store time so the join timer is accurate
-            self.updateStamp()
-            self.join(uid=remote.uid, cascade=True, timeout=30)
+            while True:
+                name = randomString(8)
+                if name in self.registry: continue
+                else: break
+
+            self.registry[name] = ha
+            self.connect(name)
```
Comment (jasonalaw) [inline sovrin_client/agent/endpoint.py]: It's tough to see from the code here if it actually fixes a problem. We need at least one test of this functionality that fails before the fix and works after the fix.

**Comment by jasonalaw** [comment]
@khagesh please see my comments. Needs a test.

File: sovrin_client/agent/endpoint.py (hunk 3 above)
Comment (jasonalaw) [inline sovrin_client/agent/endpoint.py]: Should put continue and break on different lines to be more pythonic.

**Comment by khagesh** [comment]
@jasonalaw ok

File: sovrin_client/agent/endpoint.py [hunk 4]
```diff
@@ -29,7 +29,8 @@ def __init__(self, port: int, msgHandler: Callable,
         if basedirpath:
             stackParams["basedirpath"] = basedirpath
 
-        super().__init__(stackParams, self.baseMsgHandler)
+        self.registry = {}
```
Comment (rajeshkalaria80) [inline sovrin_client/agent/endpoint.py]: I am not sure if we need to introduce this, as raet stack has a method findInRemotesByName which probably you can use to determine if a name is already used or not.

**Pull request closed without merging**